ambry-server/src/main/java/com/github/ambry/server/AmbryServerRequests.java

```diff
@@ -469,9 +469,9 @@ private AdminResponse handleHealthCheckRequest(DataInputStream requestStream, Ad
           DiskManager mountedDiskManager = entry.getValue();
 
           //Checks Disk healthchecking is enabled and
-          // if a disk isn't mounted as expected or isn't healthy then store it as part of the response
-          if (!partitionsDiskManagers.contains(mountedDiskManager)
-              || mountedDiskManager.getDiskHealthStatus() != DiskHealthStatus.HEALTHY) {
+          // if a disk manager is assigned to a partition and isn't healthy then store it as part of the response
```

Comment:
what if the partitionsDiskManager does not contain mountedDiskManager ?
The two booleans are not equal. Is that intended ?

Reply:
Great question. So we only care about disk managers that are assigned to partitions. So, as long as the disk is assigned to a partition then we care about the health. This avoids caring about backup/unused disks. The two booleans shouldn't be the same then

Reply:
ok got it